src/mono/wasm/runtime/binding_support.js

```diff
@@ -419,9 +413,8 @@ var BindingSupportLib = {
 					// TODO: Fix this once emscripten offers HEAPI64/HEAPU64 or can return them
 					throw new Error ("int64 not available");
 				case 3: // string
-					return this.conv_string (mono_obj, false);
 				case 29: // interned string
```

Comment:
now that there is no distinction here between 3 and 29 can remove the different types and avoid the check on that side?

Reply:
I think knowing the types are different is useful because it allows us to do error checks (for operations that are only valid with an interned string). The cost of the extra branch here shouldn't be a big deal